aten/src/ATen/cpu/vec256/vec256_float_neon.h

```diff
@@ -642,6 +642,893 @@ Vec256<float> inline fmadd(const Vec256<float>& a, const Vec256<float>& b, const
+              : [res_mask] "+w" (vec.values.val[0])
+              : [in_b] "w" (b.values.val[0]), [in_a] "w" (a.values.val[0])
+              :);
+          return vec;
+        }
+      case 3:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_low = {0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0x0};
+#ifdef __clang__
+          vec.values.val[0] = (float32x4_t)(mask_low);
+#else
+          vec.values.val[0] = vreinterpretq_f32_u32(mask_low);
+#endif
+          vec.values.val[1] = a.values.val[1];
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[0])
+              : [in_b] "w" (b.values.val[0]), [in_a] "w" (a.values.val[0])
+              :);
+          return vec;
+        }
+      case 4:
+        return Vec256<float>(b.values.val[0], a.values.val[1]);
+      case 5:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_high = {0xFFFFFFFF, 0x0, 0x0, 0x0};
+          vec.values.val[0] = b.values.val[0];
+#ifdef __clang__
+          vec.values.val[1] = (float32x4_t)(mask_high);
+#else
+          vec.values.val[1] = vreinterpretq_f32_u32(mask_high);
+#endif
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[1])
+              : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+              :);
+          return vec;
+        }
+      case 6:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_high = {0xFFFFFFFF, 0xFFFFFFFF, 0x0, 0x0};
+          vec.values.val[0] = b.values.val[0];
+#ifdef __clang__
+          vec.values.val[1] = (float32x4_t)(mask_high);
+#else
+          vec.values.val[1] = vreinterpretq_f32_u32(mask_high);
 #endif
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[1])
+              : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+              :);
+          return vec;
+        }
+      case 7:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_high = {0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0x0};
+          vec.values.val[0] = b.values.val[0];
+#ifdef __clang__
+          vec.values.val[1] = (float32x4_t)(mask_high);
+#else
+          vec.values.val[1] = vreinterpretq_f32_u32(mask_high);
+#endif
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[1])
+              : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+              :);
+          return vec;
+        }
+    }
+    return b;
+  }
+  static Vec256<float> loadu(const void* ptr, int64_t count = size()) {
+    if (count == size()) {
+      volatile register float32x4_t low asm("q3");
+      volatile register float32x4_t high asm("q4");
+      __asm__ __volatile__ (
+          "vld1.32 {d6, d7, d8, d9}, [%[in_ptr]]\n\t"
```

Comment:
Could you please try to use `vld1.32 {%S0 - %V0}` instead of naming registers d6 to d9?
Unfortunately this will work only with gcc, and not with clang.